src/dfx-core/src/build.rs

```diff
@@ -0,0 +1,35 @@
+use std::{env, fs::File, io::Write, path::Path};
+
+const CANISTER_IDS_URL: &str = "https://raw.githubusercontent.com/dfinity/ic/1402bf35308ec9bd87356c26f7c430f49b49423a/rs/nns/canister_ids.json";
+fn define_well_known_canisters() {
+    let well_known_canisters = reqwest::blocking::get(CANISTER_IDS_URL)
+        .unwrap()
+        .error_for_status()
+        .unwrap()
+        .text()
+        .unwrap();
+
+    let out_dir = env::var("OUT_DIR").unwrap();
+    let loader_path = Path::new(&out_dir).join("well_known_canisters.rs");
+    let mut f = File::create(loader_path).unwrap();
+    f.write_all(
```

Comment:
The format of canister_ids.json is canister name -> network -> canister id.  For these well-known canisters, the canister id will be the same on every network.

So instead of including the json contents as a string and parsing json in the emitted code, I'd suggest instead doing that here and emitting code for a lazily instantiated HashMap<CanisterName, Principal>